augur/refine.py

```diff
@@ -342,6 +343,21 @@ def are_sequence_states_different(nuc1, nuc2):
         print("ERROR: divergence unit",args.divergence_units,"not supported!", file=sys.stderr)
         return 1
 
+    if args.content_aware_internal_node_names:
+        terminals = [n.name for n in T.find_clades() if n.is_terminal()]
+        terminals.sort()
+        internals = {n.name for n in T.find_clades() if not n.is_terminal()}
+        assert all(n.startswith('NODE_') for n in internals)
+        # compute hash similar to git short commit hash
+        import hashlib
+        id = hashlib.sha256("".join(terminals).encode('utf-8')).hexdigest()[0:7]
+        def rename(name):
+            if name not in internals: return name
+            return f"NODE_{id}_{name.split('_')[1]}"
```

Comment:
The hashing of the terminals of the tree feels arbitrary and not driven by the actual properties of the hash, esp. given the truncation of it. For example, two trees with the same terminals but different structures will collide. Why not simply produce unique ids for each node instead? (I [still think we should do that](https://github.com/nextstrain/auspice/issues/1036#issuecomment-622150429).)

Reply:
> For example, two trees with the same terminals but different structures will collide.

This functionality is motivated by multi-trees where terminal node names cannot be shared across trees.